internal/util/projutil/project_util.go

```diff
@@ -140,7 +152,52 @@ func getHomeDir() (string, error) {
 // under $GOPATH and returns the current directory's import path,
 // e.g: "github.com/example-inc/app-operator"
 func CheckAndGetProjectGoPkg() string {
-	gopath := MustSetGopath(MustGetGopath())
+	gopath := MustSetWdGopath(MustGetGopath())
+	return parseGoPkg(gopath)
+}
+
+// GetGoPkg returns the current directory's import path by parsing it from
+// wd if this project's repository path is rooted under $GOPATH/src, or
+// from go.mod the project uses go modules to manage dependencies.
+//
+// Example: "github.com/example-inc/app-operator"
+func GetGoPkg() string {
+	// Default to reading from go.mod, as it should usually have the (correct)
+	// package path, and no further processing need be done on it if so.
+	if _, err := os.Stat(goModFile); err == nil {
```

Comment:
@estroz Late nit but perhaps we can fix it in #1475 
`os.Stat()` can return an error due to file permissions, filesystem error.
In which case we want to fatal error out. 

We should only proceed to looking up the path from `$GOPATH` if we know for sure that `go.mod` does not exist by checking the return error via `os.IsNotExist(err)`.
```Go
if err != nil && !os.IsNotExist(err) {
    log.Fatalf("Failed to read go.mod: %v", err)
}

// Then try parsing package path from $GOPATH (set env or default).
```

Reply:
Yes this would better inform users if there are perm errors with their go.mod. I'll add this to #1475